HACK.md

```diff
@@ -71,53 +70,68 @@ When building unit-tests, try to follow:
-
-For running E2E tests for private repositories, the **TEST_WITH_PRIVATE_REPO** environment variable needs to be set to **true**.
-If the Build private repositories [examples](test/data) contain references to private repositories you don´t have access, use
-the following variables for any related modification in the examples.
+| Environment Variable               | Path                           | Description                                         |
+|------------------------------------|--------------------------------|-----------------------------------------------------|
+| `TEST_IMAGE_REPO`                  | `spec.output.image`            | Image repository for end-to-end tests               |
+| `TEST_IMAGE_REPO_SECRET`           | `spec.output.credentials.name` | Container credentials secret name                   |
+| `TEST_IMAGE_REPO_DOCKERCONFIGJSON` | _none_                         | JSON payload equivalent to `~/.docker/config.json`  |
+
+The contents of `TEST_IMAGE_REPO_DOCKERCONFIGJSON` can be obtained from [quay.io](quay.io) using a [robot account](https://docs.quay.io/glossary/robot-accounts.html). The JSON payload is for example:
+
+```json
+{
+  "auths": {
+    "quay.io": {
+      "auth": "<secret-credentials>",
+      "email": ""
+    }
+  }
+}
+```
 
-Env var | Path | Definition
---- | --- | --- |
-**TEST_PRIVATE_REPO** | _none_ | Enables running e2e tests for private repositories |
-**TEST_PRIVATE_GITHUB** | **spec.source.url** | Private URL for the samples of the form *git@github.com* |
-**TEST_PRIVATE_GITLAB** | **spec.source.url** | Private URL for the samples of the form *git@gitlab.com* |
-**TEST_SOURCE_SECRET** | **spec.source.credentials.name** | A secret containing the SSH private key for accessing the above private repository. See [ssh-authentication](https://github.com/tektoncd/pipeline/blob/master/docs/auth.md#ssh-authentication-git). The secret definition must define two annotations: `tekton.dev/git-0: github.com` and `tekton.dev/git-1: gitlab.com`  |
+When both `TEST_IMAGE_REPO_SECRET` and `TEST_IMAGE_REPO_DOCKERCONFIGJSON` are informed, a new secret is created for end-to-end tests, named `TEST_IMAGE_REPO_SECRET`. However, when `TEST_IMAGE_REPO_DOCKERCONFIGJSON` is empty, e2e tests are expecting to find a pre-existing one.
 
-To run all strategies except buildpacks-v3 and none private git repositories tests, execute:
+To execute the end-to-end tests, run:
 
 ```sh
-operator-sdk test local ./test/e2e --up-local --namespace build-examples
+make test-e2e TEST_IMAGE_REPO_DOCKERCONFIGJSON="<JSON>"
 ```
 
-To run all strategies including buildpacks-v3 and none private git repositories tests, [setup your Quay credentials](samples/buildstrategy/buildpacks-v3#try-it-) and execute:
+Currently the end-to-end tests are not run in parallel, and may take a few minutes to complete. In average you may expect up to 10 minutes to complete all test cases.
```

Comment:
Probably depends on the system's performance. I rather observed more than ten minutes even without the private git repository tests included.

Reply:
Indeed, I've reword to be less specific.